packages/babel-helper-modules/src/index.js

```diff
@@ -231,6 +246,48 @@ const reexportGetter = template(`
 `);
 
 /**
+ * Build a statement declaring a variable that contains all of the exported
+ * variable names in an object so they can easily be referenced from an
+ * export * from statement to check for conflicts.
+ */
+function buildExportNameListDeclaration(
+  programPath: NodePath,
+  metadata: ModuleMetadata,
+) {
+  const exportedVars = Object.create(null);
+  for (const [, data] of metadata.local) {
+    for (const name of data.names) {
+      exportedVars[name] = true;
+    }
+  }
+
+  let hasRexport = false;
+  for (const [, data] of metadata.source) {
+    for (const [exportName] of data.reexports) {
+      exportedVars[exportName] = true;
+    }
+    for (const exportName of data.reexportNamespace) {
+      exportedVars[exportName] = true;
+    }
+
+    hasRexport = hasRexport || data.reexportAll;
+  }
+
+  if (!hasRexport || Object.keys(exportedVars).length === 0) return null;
+
+  const name = programPath.scope.generateUidIdentifier("exportNames");
+
+  delete exportedVars.default;
```

Comment:
better to just not set it if `exportName` is `'default'`

Reply:
Eh, I really don't think in this case it makes any difference at all.